### PR TITLE
Add updated requisition types

### LINKED ISSUE #672: Requisition tsconfig is wrong and has a bunch of errors
😁

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #672 

- Maybe the nullable dates shouldn't be a thing. Looking to change this up and remove the reducer anyway, though, so.. 🤷

### CODE REVIEW

File: packages/common/src/additions.schema.graphql [hunk 1]
```diff
@@ -314,7 +314,7 @@ union RequisitionsResponse = ConnectorError | RequisitionConnector
 
 type RequisitionConnector {
   totalCount: Int!
-  nodes: [RequisitionNode]!
+  nodes: [RequisitionNode!]!
```
Comment (mark-prins) [inline packages/common/src/additions.schema.graphql:317]: huh. so the `RequisitionNode` is required and the `nodes` prop is also required? is that what this means?

File: packages/common/src/ui/components/buttons/ColorSelectButton.tsx [hunk 2]
```diff
@@ -44,7 +44,7 @@ export const ColorSelectButton: FC<ColorSelectButtonProps> = ({
         disabled={disabled}
         icon={
           <CircleIcon
-            htmlColor={color}
+            htmlColor={color ?? '##8f90a6'}
```
Comment (mark-prins) [inline packages/common/src/ui/components/buttons/ColorSelectButton.tsx:47]: are two hashes required?

**Review by mark-prins — APPROVED** [review]
👍  that all looks good to me!

File: packages/common/src/additions.schema.graphql (hunk 1 above)
Comment (joshxg) [reply packages/common/src/additions.schema.graphql:317]: Yeah.. that's how the backend does it in the schema also, I did it incorrectly for this. The type being generated was an array of `RequisitionNode | null` (and removing the final ! meant nodes: Array<RequisitionNode | null> | null).. so to get Array<RequisitionNode> needs `!` in both places

File: packages/common/src/ui/components/buttons/ColorSelectButton.tsx (hunk 2 above)
Comment (joshxg) [reply packages/common/src/ui/components/buttons/ColorSelectButton.tsx:47]: 😁

**Pull request merged**